lib/livebook_web/live/hub/edit/team_component.ex

```diff
@@ -116,11 +123,64 @@ defmodule LivebookWeb.Hub.Edit.TeamComponent do
               environment variables.
             </p>
 
-            <.code_preview
-              source_id={"offline-deployment-#{@hub.id}"}
-              source={@dockerfile}
-              language="dockerfile"
-            />
+            <div id="env-code" class="bg-gray-900 p-2 rounded-md">
+              <div class="flex justify-between items-center mb-2 ">
+                <span class="text-gray-200 font-mono px-4 text-sm"> Dockerfile </span>
+                <button
+                  class="flex items-center justify-center bg-gray-900 hover:bg-gray-200 focus:bg-white text-gray-200 hover:text-gray-700 focus:text-gray-900 rounded-md font-mono text-xs gap-1 px-1 mr-2"
+                  data-copy
+                  data-tooltip="Copied to clipboard"
+                  type="button"
+                  aria-label="copy to clipboard"
+                  phx-click={
+                    JS.dispatch("lb:clipcopy", to: "#offline-deployment-#{@hub.id}-highlight")
+                    |> JS.add_class(
+                      "tooltip top",
+                      to: "#env-code [data-copy]",
+                      transition: {"ease-out duration-200", "opacity-0", "opacity-100"}
+                    )
+                    |> JS.remove_class(
+                      "tooltip top",
+                      to: "#env-code [data-copy]",
+                      transition: {"ease-out duration-200", "opacity-0", "opacity-100"},
+                      time: 2000
+                    )
+                  }
+                >
+                  <.remix_icon icon="clipboard-line" class="text-xl" /> Copy code
+                </button>
+              </div>
+
+              <.code_preview
+                source_id={"offline-deployment-#{@hub.id}"}
```

Comment:
Also do this and it should work :)

```suggestion
                source_id={"offline-deployment-#{@hub.id}-source"}
```

Reply:
Funny, I don't know why when using -highlight, it worked even without the -highlight on the source_id.